src/main/dialogs/promotionWindowDialog.js

```diff
@@ -0,0 +1,159 @@
+                        funcName: "{promotionWindowDialog}.close"
+                    }
+                }
+            }
+        },
+        closeTimer: {
+            type: "gpii.app.timer",
+            options: {
+                listeners: {
+                    "onTimerFinished": {
+                        funcName: "{promotionWindowDialog}.close"
+                    }
+                }
+            }
+        },
+        showTimer: {
+            type: "gpii.app.timer",
+            options: {
+                listeners: {
+                    "onTimerFinished": {
+                        funcName: "{promotionWindowDialog}.show"
+                    }
+                }
+            }
+        }
+    }
+});
+
+/**
+ * Shows the promotion window dialog. The '{promotionWindowDialog}.show' function
+ * is invoked through the 'showTimer' component. If a delay is not specified
+ * the promotion dialog will be shown without delay. This function also set the position
+ * of the promotion dialog with given 'x' and 'y' offset. If offset 'x' and 'y' are not given
+ * the window will be displayed with the default position (0, 0). If the 'centered' option
+ * is specified, the dialog will be positioned in the center of the screen.
+ * @param {gpii.app.promotionWindowDialog} that - The instance of the widget.
+ * @param {gpii.app.timer} timer - An instance of `gpii.app.timer` used for showing
+ * the widget with a delay.
+ * @param {Integer} offsetX - The x offset from the right edge of the screen.
+ * @param {Integer} offsetY - The y offset from the bottom edge of the screen.
+ * @param {gpii.app.tray} tray - The `gpii.app.tray` instance.
+ */
+gpii.app.promotionWindowDialog.show = function (that, timer, offsetX, offsetY, tray) {
+    if (that.options.siteConfig.positionByTrayIcon) {
+        // getting the tray position only when it's required (siteConfig > positionByTrayIcon)
+        var trayPosition = tray.getIconBounds(),
+            screen = require("electron").screen,
+            taskBarHeight = screen.getPrimaryDisplay().bounds.height - screen.getPrimaryDisplay().workAreaSize.height,
+            scaleFactor = screen.getPrimaryDisplay().scaleFactor,
+            displaySize = { width: screen.getPrimaryDisplay().workAreaSize.width, height: screen.getPrimaryDisplay().workAreaSize.height - taskBarHeight},
+            offset = { x: displaySize.width - ((trayPosition.x + (trayPosition.width / 2)) / scaleFactor), y: displaySize.height - ((trayPosition.y - (trayPosition.height / 2)) / scaleFactor) };
+
+        // setting the position next to the tray icon
+        that.setPosition(offset.x, offset.y);
+    } else if (that.options.siteConfig.centered) {
+        // centering the promo window
+        var centeredPosition = gpii.browserWindow.computeCentralWindowPosition(that.options.siteConfig.width, that.options.siteConfig.height);
+        that.setPosition(centeredPosition.x, centeredPosition.y);
+    } else if (offsetX || offsetY) {
```

Comment:
sloppy - what if they are both numerically zero? Better to check whether they are both not undefined, or else, given this is the fallback branch, simply unconditionally do that.setPosition(offsetX || 0, offsetY || 0)